### PR TITLE
Apply shape constraints

### PR DESCRIPTION
**Pull request by bendichter**
## Checklist

- [x] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [x] Have you ensured the PR description clearly describes problem and the solution?
- [x] Is your contribution compliant with our coding style ? This can be checked running `flake8` from the source directory.
- [x] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/NeurodataWithoutBorders/pynwb/pulls) for the same change?
- [x] Have you included the relevant issue number using `#XXX` notation where `XXX` is the issue number ?

### CODE REVIEW

File: src/pynwb/data/nwb.image.yaml [hunk 1]
```diff
@@ -145,13 +143,14 @@ groups:
     quantity: '?'
   - name: field_of_view
     dtype: float32
-    doc: Width, height and depto of image, or imaged area (meters).
+    doc: Width, height and depth of image, or imaged area (meters).
     dims:
     - - width|height
-      - width|height|depth
+    - - width|height|depth
```
Comment (bendichter) [inline src/pynwb/data/nwb.image.yaml:147]:
```suggestion
      - width|height|depth
```

File: src/pynwb/data/nwb.image.yaml [hunk 2]
```diff
@@ -145,13 +143,14 @@ groups:
...
-    doc: Width, height and depto of image, or imaged area (meters).
+    doc: Width, height and depth of image, or imaged area (meters).
     dims:
     - - width|height
       - width|height|depth
     quantity: '?'
     shape:
-    - 2
+    - - 2
+    - - 3
```
Comment (bendichter) [inline src/pynwb/data/nwb.image.yaml:151]:
```suggestion
```

File: src/pynwb/data/nwb.image.yaml [hunk 3]
```diff
@@ -145,13 +143,13 @@ groups:
...
     dtype: float32
-    doc: Width, height and depto of image, or imaged area (meters).
+    doc: Width, height and depth of image, or imaged area (meters).
     dims:
     - - width|height
       - width|height|depth
     quantity: '?'
     shape:
-    - 2
+    - - 2
```
Comment (bendichter) [inline src/pynwb/data/nwb.image.yaml:150]:
```suggestion
    - 2
```

File: src/pynwb/ophys.py [hunk 4]
```diff
@@ -107,14 +107,14 @@ class TwoPhotonSeries(ImageSeries):
 
     @docval({'name': 'name', 'type': str, 'doc': 'The name of this TimeSeries dataset'},
             {'name': 'imaging_plane', 'type': ImagingPlane, 'doc': 'Imaging plane class/pointer.'},
-            {'name': 'data', 'type': ('array_data', 'data', TimeSeries),
+            {'name': 'data', 'type': ('array_data', 'data', TimeSeries), 'shape': ([None] * 3, [None] * 4),
```
Comment (ajtritt) [inline src/pynwb/ophys.py:109]: Should the analogous shape validation be added to PlaneSegmentation.add_roi?

**Review by ajtritt — DISMISSED** [review]
Looks good to me. I only had one question... I'll leave it up to your judgement.

**Pull request merged**